MbinOrg/mbin
# feat: implement private mode

Implement a private mode, with associated settings and environment variables.
When private mode is enabled, unauthenticated users cannot view any content of the instance, they MUST be authenticate.

References #577 

## How to test

- Start an instance with the `KBIN_PRIVATE_INSTANCE` env variable set to false, you should be able to navigate everything in your instance, wether you are authenticated or not
- Restart your instance with the `KBIN_PRIVATE_INSTANCE` env variable set to true, you should be redirected to the login form, whatever you are trying to do. **Registration, password reset, email activation, consent should still be available even in private mode**.
- An entry should be available on the admin settings page, you can also play with it
